src/renameKey.js

```diff
@@ -0,0 +1,36 @@
+import { always, curry, equals, when } from 'ramda';
+
+import renameKeysWith from './renameKeysWith';
+
+/**
+ * Creates a new object with the own properties of the provided object, but a
```

Comment:
May I suggest the following wording? It introduces less abstract terms and IMHO reads clearer

```js
/**
 * Creates a new object with the own properties of the provided object, but a
 * single key is renamed from `oldKey` to `newKey`.
 *
 * Keep in mind that in the case of keys conflict is behavior undefined and
 * the result may vary between various JS engines!
```